### PR TITLE
bpo-33463: _asdict --> dict, not OrderedDict

### PR DESCRIPTION
**Pull request by selik**
Now that dict is tracking insertion order, namedtuple._asdict can be
switched to returning a basic dict instead of an OrderedDict. The
interface is largely similar, but some minor differences will require a
deprecation cycle before making the change.

https://bugs.python.org/issue33463

### CODE REVIEW

**Comment by selik** [comment]
I changed the docs to note the change as of Version 3.8. If this is accepted, I'll create separate pull requests to add warnings to the docs of older versions.

File: Doc/library/collections.rst [hunk 1]
```diff
@@ -886,18 +886,21 @@ field names, the method and attribute names start with an underscore.
 
 .. method:: somenamedtuple._asdict()
 
-    Return a new :class:`OrderedDict` which maps field names to their corresponding
+    Return a new :class:`dict` which maps field names to their corresponding
```
Comment (serhiy-storchaka) [inline Doc/library/collections.rst]: Maybe "Return a new ordered dictionary ..."?

File: Doc/library/collections.rst [hunk 2]
```diff
@@ -886,18 +886,21 @@ field names, the method and attribute names start with an underscore.
...
         >>> p = Point(x=11, y=22)
         >>> p._asdict()
-        OrderedDict([('x', 11), ('y', 22)])
+        {'x': 11, 'y': 22}
 
     .. versionchanged:: 3.1
         Returns an :class:`OrderedDict` instead of a regular :class:`dict`.
 
+    .. versionchanged:: 3.8
+        Returns an :class:`dict` instead of a regular :class:`OrderedDict`.
```
Comment (serhiy-storchaka) [inline Doc/library/collections.rst]: Maybe a clarification is needed.

Why the "regular" adjective is used with OrderedDict.

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]
Needed a test.

Please add a What's New entry and remove a duplicated news entry, and clarify the reason of reverting the 3.1 change.

**Comment by serhiy-storchaka** [comment]
For testing just add something like:
```python
self.assertEqual(list(p._asdict().items()), [('x': 11), ('y': 22)])
```
`dict` equality doesn't take the order into account.

Since this change breaks compatibility, it needs to be mentioned in the What's New document, in the "Porting to 3.8" section.

File: Doc/library/collections.rst (hunk 2 above)
Comment (merwok) [reply Doc/library/collections.rst]: an dict → a dict

**Comment by selik** [comment]
I added a bit to What's New, but I just realized I need to move that to the "Porting to 3.8" section. I'll do that shortly.

**Comment by rhettinger** [comment]
Thanks for the PR. 

Have applied an alternative version so the issue is taken care of.

**Pull request closed without merging**